src/HDF5.jl

```diff
@@ -1262,7 +1263,7 @@ unpad(s, pad::Integer) = unpad(String(s), pad)
 
 # Dereference
 function getindex(parent::Union{File, Group, Dataset}, r::Reference)
-    r.r == HOBJ_REF_T_NULL && error("Reference is null")
+    r == Reference() && error("Reference is null")
```

Comment:
Is it normal for a no arg constructor to have by default a NULL reference? I suppose it makes sense for this case, since by definition if there is nothing to reference it will be a null reference.

Reply:
I was thinking of it more in terms as providing a default zero-initialized data structure. It just so happens to be useful to compare against to indicate whether a reference is valid or not.